packages/addons/addon-depends/jre-depends/jre-libbluray/package.mk

```diff
@@ -0,0 +1,42 @@
+# SPDX-License-Identifier: GPL-2.0
+# Copyright (C) 2018-present Team LibreELEC (https://libreelec.tv)
+# Copyright (C) 2019-present vpeter
+
+. $(get_pkg_directory libbluray)/package.mk
+
+PKG_NAME="jre-libbluray"
+PKG_DEPENDS_TARGET+=" jdk-x86_64-zulu apache-ant:host"
+PKG_LONGDESC="libbluray jar for BD-J menus"
+PKG_URL=""
+
+unpack() {
+  mkdir -p $PKG_BUILD
+  tar --strip-components=1 -xf $SOURCES/${PKG_NAME:4}/${PKG_NAME:4}-$PKG_VERSION.tar.bz2 -C $PKG_BUILD
+
+  # use patches from original package folder
+  local pkg_orig=$(get_pkg_directory libbluray)
+  local pkg_copy=$(get_pkg_directory ${PKG_NAME})
+
+  pkg_orig=${pkg_orig#"${ROOT}/"}
+  pkg_copy=${pkg_copy#"${ROOT}/"}
+  pkg_copy=$(echo "${pkg_copy}" | sed 's|[a-zA-Z0-9_\-]*|..|g')
+
+  PKG_PATCH_DIRS="${pkg_copy}/../${pkg_orig}/patches"
```

Comment:
Yikes. Alternatively you could just add a `patches` sym link from this package to `libbluray/patches`? But that would preclude in future the addition of any patches that are specific to this package.

Reply:
Why copying patches around and taking care on changes if there is no reason to do that? If some other package can be sourced like in this case then it should also care to apply patches from it. My solution is not perfect but works just fine. In case there are local patches this second folder can be added to PKG_PATCH_DIRS and it will be applied also.

Reply:
We could make this a lot less fuggly by adding support for absolute paths when defining PKG_PATCH_DIRS, what do you think? Then you can just add:
```
PKG_PATCH_DIRS="$(get_pkg_directory libbluray)/patches"
```

Reply:
Try this: http://ix.io/1NMz with the absolute path change above.

Reply:
I think it works perfectly. Will run few more tests.

Reply:
Just looking at the `scripts/unpack` code, and you may need to define `PKG_PATCH_DIRS` _outside_ of `unpack()` because of this: https://github.com/LibreELEC/LibreELEC.tv/blob/master/scripts/unpack#L69-L72 where `unpack` is called from within a sub-process so anything you set within `unpack()` will be forgotten on the return to `scripts/unpack`.

Basically, just define your `PKG_PATCH_DIRS` in the "global" section instead, not within `unpack()`.

Reply:
Your existing code may have worked with `libreelec-9.0`, but it wouldn't have worked with 9.2/master even without absolute path support as `PKG_PATCH_DIRS` would always be blank.

Reply:
I wonder if I should remove the sub-process around the `unpack()` call, as it's potentially a trap due to the unusual behaviour...

Reply:
https://github.com/LibreELEC/LibreELEC.tv/compare/master...MilhouseVH:le10_abs_patch_dir - removes the sub-process, although I still think you should define `PKG_PATCH_DIRS` in the global section!